### PR TITLE
Convert SGv2 REST API from DropWizard to Quarkus

### LINKED ISSUE #1892: Convert SGv2/REST API to Quarkus
After initial stand-alone Stargate V2 REST API has been successfully deployed, we should convert it from DropWizard to Quarkus (see #1526).

This issue is the main placeholder to be referenced from specific task issues, to be filed once investigation and work starts.

Note that there is an immediate question of how to change build (or perhaps repo structure) to solve the issued of JDK-requirement between Quarkus (at least 11 but ideally 17) and Persistence/OSGi (Java 8 only).

### PR DESCRIPTION
**Pull request by tatu-at-datastax**
**What this PR does**:

* All existing REST endpoints (data and schema) converted to Quarkus, still using (blocking) JAX-RS
* Main Integration tests (Schema/Keyspaces) converted, running as dockerized (non-JVM) tests
     * Schema tests
     * All Rows (non-schema) ITs
     * SAI, MV tests (non-C*4)
     * Basic healthcheck/metrics endpoint access
* Still to convert (follow-up)
     * DSE-specific `RestApiv2DseTest`
* Removed after conversion:
    * Main non-Quarkus SGv2/REST tests (`RestApiv2RowsTest`, `RestApiv2SchemaTest`) removed
    * SAI, MV tests

As follow-up need to:

* Remove old SGv2 implementation, code to build
    * Can't remove before ensuring Q-based implementation fully works, including CI

**Which issue(s) this PR fixes**:
Fixes #1892

**Checklist**
- [x] Changes manually tested
- [x] Automated Tests added/updated
- [x] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: apis/sgv2-restapi/src/main/java/io/stargate/sgv2/restapi/service/resources/schemas/Sgv2UDTsResourceImpl.java [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+      return restServiceError(
+          Response.Status.BAD_REQUEST,
+          "addFields and/or renameFields is required to update an UDT");
+    }
+
+    if (addFields != null && !addFields.isEmpty()) {
+      List<Column> columns = columns2columns(addFields);
+      QueryOuterClass.Query query =
+          new QueryBuilder().alter().type(keyspaceName, typeName).addColumn(columns).build();
+      /*QueryOuterClass.Response grpcResponse =*/ bridge.executeQuery(query);
```
Comment (EricBorczuk) [inline apis/sgv2-restapi/src/main/java/io/stargate/sgv2/restapi/service/resources/schemas/Sgv2UDTsResourceImpl.java:181]: Same here, should this comment be removed?

File: apis/sgv2-restapi/src/main/resources/banner/banner.txt [hunk 2]
```diff
@@ -0,0 +1,3 @@
+============================================================================
+ Ad Space For Sale!!!
```
Comment (EricBorczuk) [inline apis/sgv2-restapi/src/main/resources/banner/banner.txt:2]: :O

File: apis/sgv2-restapi/src/test/java/io/stargate/sgv2/it/RestApiV2QIntegrationTestBase.java [hunk 3]
```diff
@@ -0,0 +1,633 @@
...
+    row.put("id", 2);
+    row.put("firstName", "Jane");
+    row.put("expense_id", 1);
+    insertRow(keyspaceName, tableName, row);
+
+    return firstRowId;
+  }
+
+  protected void setupMixedClusteringTestCase(String keyspaceName, String tableName) {
+    // createTestTableWithMixedClustering(keyspaceName, tableName)
```
Comment (EricBorczuk) [inline apis/sgv2-restapi/src/test/java/io/stargate/sgv2/it/RestApiV2QIntegrationTestBase.java:590]: Should this be removed

File: apis/sgv2-restapi/src/test/java/io/stargate/sgv2/it/RestApiV2QRowGetIT.java [hunk 4]
```diff
@@ -0,0 +1,948 @@
...
+            .queryParam("where", whereClause)
+            .when()
+            .get(path)
+            .then()
+            .statusCode(HttpStatus.SC_BAD_REQUEST)
+            .extract()
+            .asString();
+    ApiError error = readJsonAs(response, ApiError.class);
+    assertThat(error.code()).isEqualTo(HttpStatus.SC_BAD_REQUEST);
+    assertThat(error.description()).contains("Invalid 'where' parameter, problem: ");
```
Comment (EricBorczuk) [inline apis/sgv2-restapi/src/test/java/io/stargate/sgv2/it/RestApiV2QRowGetIT.java:172]: Maybe this could also check on what the problem is?

**Review by EricBorczuk — APPROVED** [review]
This looks good to me overall!

I had a few trivial things, but nothing blocking

File: apis/sgv2-restapi/src/main/java/io/stargate/sgv2/restapi/service/resources/Sgv2RowsResourceImpl.java [hunk 5]
```diff
@@ -0,0 +1,543 @@
...
+
+  @Override
+  public Response createRow(
+      final String keyspaceName, final String tableName, final String payloadAsString) {
+    requireNonEmptyKeyspaceAndTable(keyspaceName, tableName);
+    Map<String, Object> payloadMap;
+    try {
+      payloadMap = parseJsonAsMap(payloadAsString);
+    } catch (Exception e) {
+      throw invalidPaylodException(e);
```
[the remark this answers is not in the document]
Comment (tatu-at-datastax) [reply apis/sgv2-restapi/src/main/java/io/stargate/sgv2/restapi/service/resources/Sgv2RowsResourceImpl.java:184]: ah. Thanks! :)

File: apis/sgv2-restapi/src/main/java/io/stargate/sgv2/restapi/service/resources/schemas/Sgv2KeyspacesResourceImpl.java [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+    // No real contents; can ignore ResultSet it seems and only worry about exceptions
+
+    final Map<String, Object> responsePayload = Collections.singletonMap("name", keyspaceName);
+    return Response.status(Status.CREATED).entity(responsePayload).build();
+  }
+
+  @Override
+  public Response deleteKeyspace(final String keyspaceName) {
+    Query query = new QueryBuilder().drop().keyspace(keyspaceName).ifExists().build();
+    /*QueryOuterClass.Response grpcResponse =*/ bridge.executeQuery(query);
```
[the remark this answers is not in the document]
Comment (tatu-at-datastax) [reply apis/sgv2-restapi/src/main/java/io/stargate/sgv2/restapi/service/resources/schemas/Sgv2KeyspacesResourceImpl.java:126]: Yes, I should just remove it, thanks!

File: apis/sgv2-restapi/src/main/resources/banner/banner.txt (hunk 2 above)
Comment (tatu-at-datastax) [reply apis/sgv2-restapi/src/main/resources/banner/banner.txt:2]: :)

Will change.

File: apis/sgv2-restapi/src/test/java/io/stargate/sgv2/it/RestApiV2QIntegrationTestBase.java (hunk 3 above)
Comment (tatu-at-datastax) [reply apis/sgv2-restapi/src/test/java/io/stargate/sgv2/it/RestApiV2QIntegrationTestBase.java:590]: Sure.

File: apis/sgv2-restapi/src/test/java/io/stargate/sgv2/it/RestApiV2QRowGetIT.java (hunk 4 above)
Comment (tatu-at-datastax) [reply apis/sgv2-restapi/src/test/java/io/stargate/sgv2/it/RestApiV2QRowGetIT.java:172]: I was thinking of that, but I think the actual description seemed gRPC-specific and wasn't sure if to include it. But let me go back and double-check, either adding a comment or added check.

File: apis/sgv2-restapi/README.md [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+## Configuration properties
+
+There are two main configuration property prefixes used, `stargate.` and `quarkus.`.
+Latter covers configuration of Quarkus-provided standard features, former Stargate-specific features.
+
+(TO BE COMPLETED)
+
+### Configuration: General Quarkus properties
+
+(TO BE WRITTEN)
```
[the remark this answers is not in the document]
Comment (tatu-at-datastax) [reply apis/sgv2-restapi/README.md:43]: I should do something here, but I think there is also some discussion on where to add shared/common properties.
I'll have a look at what GraphQL/Docs API do.

File: apis/sgv2-restapi/src/test/java/io/stargate/sgv2/it/RestApiV2QRowGetIT.java (hunk 4 above)
Comment (tatu-at-datastax) [reply apis/sgv2-restapi/src/test/java/io/stargate/sgv2/it/RestApiV2QRowGetIT.java]: There is a real error message, will check that. Just need to see if it's persistence-dependant (different on C*3 vs c*4 etc).

Comment (tatu-at-datastax) [reply apis/sgv2-restapi/src/test/java/io/stargate/sgv2/it/RestApiV2QRowGetIT.java]: aaand of course it is slightly different. So need to tune the match a bit.

File: apis/sgv2-restapi/README.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (tatu-at-datastax) [reply apis/sgv2-restapi/README.md]: Removed these sections for now; will need to update README properly with follow up.

**Pull request merged**